### ISSUE TITLE
BoundsError in Sparspak on CI

### ISSUE THREAD

**Issue by fredrikekre** [SciML/LinearSolve.jl#258]
See https://github.com/SciML/LinearSolve.jl/actions/runs/3967886786/jobs/6800368191

```
ERROR: LoadError: BoundsError: attempt to access 3-element Vector{Int64} at index [4]
Stacktrace:
  [1] getindex
    @ ./array.jl:924 [inlined]
  [2] isstructuresymmetric(g::Sparspak.SpkGraph.Graph{Int64})
    @ Sparspak.SpkGraph ~/.julia/packages/Sparspak/uIYt5/src/Graph/SpkGraph.jl:336
  [3] makestructuresymmetric(g::Sparspak.SpkGraph.Graph{Int64})
    @ Sparspak.SpkGraph ~/.julia/packages/Sparspak/uIYt5/src/Graph/SpkGraph.jl:120
  [4] _findorder!(s::Sparspak.SpkSparseBase._SparseBase{Int64, Float64}, orderfunction::typeof(Sparspak.SpkMmd.mmd!))
    @ Sparspak.SpkSparseBase ~/.julia/packages/Sparspak/uIYt5/src/SparseMethod/SpkSparseBase.jl:178
  [5] _findorder!
    @ ~/.julia/packages/Sparspak/uIYt5/src/SparseMethod/SpkSparseBase.jl:184 [inlined]
  [6] findorder!
    @ ~/.julia/packages/Sparspak/uIYt5/src/SparseMethod/SpkSparseSolver.jl:120 [inlined]
  [7] sparspaklu!(lu::Sparspak.SpkSparseSolver.SparseSolver{Int64, Float64}, m::SparseArrays.SparseMatrixCSC{Float64, Int64})
    @ Sparspak.SparseCSCInterface ~/.julia/packages/Sparspak/uIYt5/src/SparseCSCInterface/SparseCSCInterface.jl:247
  [8] #solve#18
    @ ~/work/LinearSolve.jl/LinearSolve.jl/src/factorization.jl:522 [inlined]
```

cc @j-fu @PetrKryslUCSD

**Comment by j-fu** [comment]
I can reproduce this at my computer, however, the error occurs randomly, i.e. sometimes it shows up, sometimes not...

First time seeing this coming up, other ways to work with sparse random matrices so far didn't show this behavior.

For the time being I propose to remove sparspak from precompilation (this is where the error occurs). I think it can be kept in the package tests, as there, different matrices are used.

Will try to find time over the WE to investigate deeper.

**Comment by ChrisRackauckas** [comment]
Removed from precompilation for now.

**Comment by PetrKryslUCSD** [comment]
I think I found an error in the original fortran code: the graph field adj was being accessed
beyond the end of the array. I added a check on line https://github.com/PetrKryslUCSD/Sparspak.jl/blob/958744abe7f5a97a52419b928541bd4623c6f182/src/Graph/SpkGraph.jl#L336. All tests pass now.

**Comment by PetrKryslUCSD** [comment]
Spoke too soon: Not good. Sparspak now fails in simpletest2.

**Comment by j-fu** [comment]
Fixed with 710fc37 via https://github.com/PetrKryslUCSD/Sparspak.jl/commit/2b5b685895abdf693a7f72f5903ff87be0bc83b5
